### PR TITLE
add function to list all cpus num in lists

### PR DESCRIPTION
**Pull request by kylazhang**
I need to use it in several test files so make it public here

### CODE REVIEW

File: client/base_utils.py [hunk 1]
```diff
@@ -408,6 +408,13 @@ def count_cpus():
     return cpus
 
 
+def get_cpu_lists():
+    """Get all CPUs list in local machine according to /proc/cpuinfo"""
+    cpus_result = utils.run("x=$(cat /proc/cpuinfo |grep processor|cut -d: -f2);echo $x")
```
Comment (yumingfei) [inline client/base_utils.py]: Too long, you can add one more step to solve it:
cmd = "x=..."

Comment (xutian) [inline client/base_utils.py]: @kylazhang
your cmd eq "awk -F: '/processor/{print $2}' /proc/cpuinfo"

**Comment by rbbratta** [comment]
We already have this function, look further in the file.

```python
def cpu_online_map():
    """
    Check out the available cpu online map
    """
    cpus = []
    for line in open('/proc/cpuinfo', 'r').readlines():
        if line.startswith('processor'):
            cpus.append(line.split()[2])  # grab cpu number
    return cpus
```


Here is a better version

```python
import re

def cpu_online_map():
    cpuinfo_file = open("/proc/cpuinfo")
    try:
        cpuinfo = cpuinfo_file.read()
    finally:
        cpuinfo_file.close()
    return [int(c) for c in re.findall(r"^processor\s+:\s+(\d+)$", cpuinfo, re.M)]
```

Note: cpuinfo lists only online cpus.

We can also simplify count_cpus()

```python
def count_cpus():
   return os.sysconf("SC_NPROCESSORS_ONLN")
```

**Comment by kylazhang** [comment]
Thanks all, I will close it and use existing function.

**Pull request closed without merging**